rust/agama-lib/src/lib.rs

```diff
@@ -41,13 +41,34 @@ mod store;
 pub use store::Store;
 pub mod questions;
 use crate::error::ServiceError;
+use reqwest::cookie::Jar;
+use reqwest::{header, Client};
+use std::sync::Arc;
 
 const ADDRESS: &str = "unix:path=/run/agama/bus";
 
 pub async fn connection() -> Result<zbus::Connection, ServiceError> {
     connection_to(ADDRESS).await
 }
 
+pub fn http_client(token: String) -> Result<reqwest::Client, ServiceError> {
```

Comment:
np: It is fine as it is. But I would consider defining our own `HTTPClient` on top of `reqwest::Client` (using the [new type idiom](https://doc.rust-lang.org/rust-by-example/generics/new_types.html)) to hide the fact that we are using reqwest. But it just a matter of personal taste. What do you think?

Reply:
Do not see its benefits

Reply:
As said, we can leave it as it is. Benefits? Well, 1) encapsultation and 2) defining our own simplified API). We follow a similar approach in the JavaScript side. But let's keep it as it is until the benefits become more clear.